src/libxtp/orbitals.cc

```diff
@@ -109,6 +136,21 @@ Eigen::MatrixXd Orbitals::DensityMatrixGroundState() const {
   return dmatGS;
 }
 
+// Density matrix for a single KS orbital
+Eigen::MatrixXd Orbitals::DensityMatrixKSstate(const QMState& state) const {
+  if (!hasMOs()) {
+    throw std::runtime_error("Orbitals file does not contain MO coefficients");
+  }
+  if (state.Type() != QMStateType::KSstate &&
```

Comment:
should be an ||

Reply:
No, it reads a bit difficult, but it says: if it is *not* a KS state and *not* a PQP state then it will not return a KS density (and print an error). So if it is a KS state or a PQP state then it will return the KS density.

Reply:
true